source/m2mresourceinstance.cpp

```diff
@@ -124,21 +124,19 @@ M2MResourceInstance::ResourceType M2MResourceInstance::resource_instance_type()
 
 bool M2MResourceInstance::handle_observation_attribute(char *&query)
 {
-    tr_debug("M2MResourceInstance::handle_observation_attribute()");
+    tr_debug("M2MResourceInstance::handle_observation_attribute - is_under_observation(%d)", is_under_observation());
     bool success = false;
     M2MReportHandler *handler = M2MBase::report_handler();
     if (handler) {
         success = handler->parse_notification_attribute(query,
                 M2MBase::base_type(), _resource_type);
-        if (success) {
-            if ((handler->attribute_flags() & M2MReportHandler::Cancel) == 0) {
+        if(success) {
+            if (is_under_observation()) {
                 handler->set_under_observation(true);
-            } else {
-                handler->set_under_observation(false);
             }
-        }
-        else {
-            handler->set_default_values();
+            else {
```

Comment:
Still the same issue, else condition is for is_under_observation() and not for if(success)